opm/parser/eclipse/Parser/MessageContainer.hpp

```diff
@@ -0,0 +1,85 @@
+/*
+  Copyright 2016 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef MESSAGECONTAINER_H
+#define MESSAGECONTAINER_H
+
+#endif // OPM_MESSAGECONTAINER_HEADER_INCLUDED
+
+#include <string>
+#include <vector>
+#include <memory>
+
+namespace Opm {
+
+    namespace MessageType {
+
+        enum MessageTypeEnum {
+            DEBUG     = 1,
```

Comment:
This particular name (`DEBUG`) conflicts with the [macro](https://github.com/OPM/opm-parser/blob/master/CMakeLists.txt#L82) that is defined in the top-level `CMakeLists.txt` in debug name.  The proper resolution is **probably** to not define the `DEBUG` macro in the build system.

As of right now, module opm-parser does not build in debug mode due to this conflict.